src/app/app-common/PageSource/PlainTextPageSource.cpp

```diff
@@ -246,6 +284,7 @@ void PlainTextPageSource::SetPlaceholderText(std::string_view text) {
 void PlainTextPageSource::PushMessage(std::string_view message) {
   std::unique_lock lock(mMutex);
   mMessages.push_back(std::string(message));
```

Comment:
the explicit string_view -> string constructor was required by the standards committee to make it clear there's a copy going on. Given we're keeping `mAllMessages` 'forever', we should:

- push the `std::string` to `mAllMessages` first
- change `mMessages` to be a collection of `std::string_view`, derived from `mAllMessages.back()` (not the parameter)

Reply:
Changing mMessages to string_view causes issues in LayoutMessages() where it is expected to be able to replaces \t with fixed spaces. Not sure what the best way to handle this is if we're determined to avoid copies.

Reply:
It can either be copied then, or converted to hstring/wstring first then do the replacement with wchars

Reply:
I found that if we convert the tabs once, then copying isn't required, and that didn't seem to quite fit there anyway.